db/models/user.js

```diff
@@ -2,27 +2,48 @@
 
 // bcrypt docs: https://www.npmjs.com/package/bcrypt
 const bcrypt = require('bcryptjs')
-    , {STRING, VIRTUAL} = require('sequelize')
+    , {BOOLEAN, STRING, VIRTUAL} = require('sequelize')
 
 module.exports = db => db.define('users', {
-  name: STRING,
+  userName: STRING,
   email: {
     type: STRING,
     validate: {
       isEmail: true,
-      notEmpty: true,
     }
   },
+  firstName: STRING,
+  lastName: STRING,
+  shippingAddress: STRING ,
+  creditInfo: {
+    type: INTEGER,
+    validate: {
+      isCreditCard: true
+    }
+  },
+  isAdmin: {
+    type: BOOLEAN,
+    defaultValue: false,
+  },
 
   // We support oauth, so users may or may not have passwords.
   password_digest: STRING, // This column stores the hashed password in the DB, via the beforeCreate/beforeUpdate hooks
   password: VIRTUAL // Note that this is a virtual, and not actually stored in DB
 }, {
   indexes: [{fields: ['email'], unique: true}],
   hooks: {
-    beforeCreate: setEmailAndPassword,
+    beforeCreate: setEmailAndPassword, checkGuest
     beforeUpdate: setEmailAndPassword,
   },
+  getterMethods: {
+    // checks if user is a guest or not
+    isGuest () {
+      return this.email ? false : true
```

Comment:
If we want to check this way, may want to set default value of the email column to something definitely falsey. Right now it doesn't have a default value -- am concerned it may accidentally become an empty string